internal/test/e2e/kubectl.go

```diff
@@ -165,3 +177,100 @@ func NamespaceForTest(t *testing.T, kubectl *kubectlContext) string {
 
 	return name
 }
+
+type jobBuilder struct {
```

Comment:
what's the advantage of the builder here? over an interface more like `exec.Command`?

Reply:
Yeah, I may have got carried away there.
I started out trying to simplify the creation of a custom Job resource and conflated it with deploying, waiting and getting the output of the job.
I'll simplify it.

Reply:
I've ditched the builder stuff and used a simple, single function instead.
We can extract the separate Start, Wait, Output methods if and when we need it.